src/dash-table/utils/actions.js

```diff
@@ -53,23 +53,31 @@ export const clearSelection = {
     selected_cells: []
 };
 
-export function editColumnName(column, columns, headerRowIndex) {
+export function changeColumnHeader(column, columns, headerRowIndex, mergeDuplicateHeaders, newColumnName) {
+    let cloneColumn = R.mergeRight({}, column);
+    const maxLength = getHeaderRows(columns);
+    if (typeof column.name === 'string' && maxLength > 1) {
+            const newColumnNames = Array(maxLength).fill(column.name);
+            cloneColumn = R.mergeRight(column, {name: newColumnNames});
+    }
+    const transformedColumns =  columns.slice(0);
+    const columnIndex = columns.findIndex(col => col.id === column.id);
+    transformedColumns[columnIndex] = cloneColumn;
     const { groupIndexFirst, groupIndexLast } = getGroupedColumnIndices(
-        column, columns, headerRowIndex
+        column, columns, headerRowIndex, mergeDuplicateHeaders
     );
-    /* eslint no-alert: 0 */
-    const newColumnName = window.prompt('Enter a new column name');
-    let newColumns = R.clone(columns);
+    let newColumns = R.clone(transformedColumns);
```

Comment:
I don't think the deep clone (https://ramdajs.com/docs/#clone) is necessary here. It will actually make multiple `memoizeOne` functions fail their test on their `column` argument, forcing a significant portion of the table to re-evaluate itself.